code/datums/components/food/decomposition.dm

```diff
@@ -73,6 +73,9 @@
 	if(locate(/obj/machinery/conveyor) in get_turf(food)) // Makes sure no decals spawn on disposals conveyors
 		remove_timer()
 		return
+	if(locate(/obj/structure/closet) in get_turf(food)) //Is it on a closet/fridge/locker/crate?
+		remove_timer()
+		return
```

Comment:
Remove all these stupid checks and do what I asked them to do from the beginning.

Loop over contents of the turf *once*, and check if they're in a typecache.

We should not be performing *four* `locate in turf` checks, and I don't even know why we are doing 3.

Reply:
I knew someone was going to suggest doing this. I'll do it then.

Reply:
Acomplished. Let me know if this is the incorrect way to do this.